src/components/coin.ts

```diff
@@ -150,32 +151,17 @@ export const getCurrentCoinLeaderboard = async (limit = 10): Promise<UserCoinEnt
   const db = await openDB();
   const res = await db.all(
     `
-      SELECT user_id, balance
+      SELECT user_id, balance,
+      Rank() OVER (ORDER BY balance DESC) AS rank
       FROM user_coin
-      ORDER BY balance DESC
+      ORDER BY rank ASC
```

Comment:
Once again, since rank is now manually computed, the previous simpler logic seems to be fine?

Reply:
Fixed it in this commit